genai.go

```diff
@@ -0,0 +1,382 @@
+	Url              string        `json:"url,omitempty"`
+	Usecases         []string      `json:"usecases,omitempty"`
+	Uuid             string        `json:"uuid,omitempty"`
+	Version          *ModelVersion `json:"version,omitempty"`
+}
+
+type Agreement struct {
+	Description string `json:"description,omitempty"`
+	Name        string `json:"name,omitempty"`
+	Url         string `json:"url,omitempty"`
+	Uuid        string `json:"uuid,omitempty"`
+}
+
+type ModelVersion struct {
+	Major int `json:"major,omitempty"`
+	Minor int `json:"minor,omitempty"`
+	Patch int `json:"patch,omitempty"`
+}
+
+type AgentCreateRequest struct {
+	AnthropicKeyUuid  string   `json:"anthropic_key_uuid,omitempty"`
+	Description       string   `json:"description,omitempty"`
+	Instruction       string   `json:"instruction,omitempty"`
+	KnowledgeBaseUuid []string `json:"knowledge_base_uuid,omitempty"`
+	ModelUuid         string   `json:"model_uuid,omitempty"`
+	Name              string   `json:"name,omitempty"`
+	OpenAiKeyUuid     string   `json:"open_ai_key_uuid,omitempty"`
+	ProjectId         string   `json:"project_id,omitempty"`
+	Region            string   `json:"region,omitempty"`
+	Tags              []string `json:"tags,omitempty"`
+}
+
+type AgentUpdateRequest struct {
+	AnthropicKeyUuid string   `json:"anthropic_key_uuid,omitempty"`
+	Description      string   `json:"description,omitempty"`
+	Instruction      string   `json:"instruction,omitempty"`
+	K                int      `json:"k,omitempty"`
+	MaxTokens        int      `json:"max_tokens,omitempty"`
+	ModelUuid        string   `json:"model_uuid,omitempty"`
+	Name             string   `json:"name,omitempty"`
+	OpenAiKeyUuid    string   `json:"open_ai_key_uuid,omitempty"`
+	ProjectId        string   `json:"project_id,omitempty"`
+	RetrievalMethod  string   `json:"retrieval_method,omitempty"`
+	Region           string   `json:"region,omitempty"`
+	Tags             []string `json:"tags,omitempty"`
+	Temperature      float64  `json:"temperature,omitempty"`
+	TopP             float64  `json:"top_p,omitempty"`
+	Uuid             string   `json:"uuid,omitempty"`
+}
+
+// List returns a list of Gen AI Agents
+func (s *AgentServiceOp) ListAgents(ctx context.Context, opt *ListOptions) ([]*Agent, *Response, error) {
+	path, err := addOptions(agentConnectBasePath, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(genAIAgentsRoot)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+	if l := root.Links; l != nil {
+		resp.Links = l
+	}
+	if m := root.Meta; m != nil {
+		resp.Meta = m
+	}
+	return root.Agents, resp, nil
+}
+
+// Create creates a new Gen AI Agent by providing the AgentCreateRequest object
+func (s *AgentServiceOp) CreateAgent(ctx context.Context, create *AgentCreateRequest) (*Agent, *Response, error) {
+	path := agentConnectBasePath
+
```

Comment:
```suggestion
	if create.ProjectId == "" {
		return nil, nil, fmt.Errorf("Project ID is required")
	}
	if create.Region == "" {
		return nil, nil, fmt.Errorf("Region is required")
	}
	if create.Instruction == "" {
		return nil, nil, fmt.Errorf("Instruction is required")
	}
	if create.ModelUuid == "" {
		return nil, nil, fmt.Errorf("ModelUuid is required")
	}
	if create.Name == "" {
		return nil, nil, fmt.Errorf("Name is required")
	}
```

Let's just add these checks before the API call, since the API will fail with out them.